mgmt: mcumgr: grp: os_mgmt: Add datetime get/set functions

nordicjm opened the pull request:
Adds datetime set and get functions which allow for setting and getting the current time to/from the rtc alias device

nordicjm commented:
@bjarki-trackunit might be of interest

bjarki-trackunit commented on subsys/mgmt/mcumgr/grp/os_mgmt/Kconfig:
```diff
@@ -135,6 +135,26 @@ config MCUMGR_GRP_OS_ECHO
+config MCUMGR_GRP_OS_DATETIME
+	bool "Support for datetime command"
+	depends on RTC
+	select MCUMGR_SMP_CBOR_MIN_DECODING_LEVEL_2
```
Adding `depends on $(dt_alias_exists,rtc)` could catch a missing rtc: alias in a nice way :)

nordicjm replied:
Nice! Fixed

nordicjm committed:
mgmt: mcumgr: grp: os_mgmt: Add datetime get/set functions

Adds datetime set and get functions which allow for setting and
getting the current time to/from the rtc alias device

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
tests: mgmt: mcumgr: Add os_mgmt_datetime test

Adds a test which tests datetime set/get and hook callbacks

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
doc: release: 3.6: Add note on MCUmgr OS mgmt datetime

Adds a note that datetime support has been added to MCUmgr

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
doc: mgmt: mcumgr: Remove OS mgmt datetime unimplemented note

This functionality is now implemented

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>
